src/node_options-inl.h

```diff
@@ -475,6 +479,12 @@ void OptionsParser<Options>::Parse(
             value = value.substr(1);  // Treat \- as escaping an -.
         }
       }
+    } else if (info.type == kBoolean) {
+        if (equals_index != std::string::npos) {
+          // Boolean options don't accept arguments
+          errors->push_back(UnexpectedArgumentErr(name));
+          break;
+        }
     }
```

Comment:
I think you can combine the ifs...

```suggestion
    } else if (info.type == kBoolean && equals_index != std::string::npos) {
      // Boolean options don't accept arguments
      errors->push_back(UnexpectedArgumentErr(name));
      break;
    }
```